clang/lib/Sema/SPIRVBuiltins.td

```diff
@@ -745,7 +745,7 @@ foreach IType = [Char, Short, Int, Long] in {
   }
 }
 
-foreach InType = TLAll.List in {
+foreach InType = TLUnsignedInts.List in {
```

Comment:
Why this restriction ? It is valid to convert a signed int to an unsigned one.

Reply:
Yes, it is valid, but SatConvertSToU exists for this purpose.

Reply:
`UConvert`/`SConvert` does not saturate the result unless decorated...

Reply:
I will check it, but I have doubt, that UConvert gets non unsigned arguments.

Reply:
When I run test that converts int to uint, it writes "call to '__spirv_UConvert_Ruint' is ambiguous", that means it make implicit conversion form signed to unsigned so that push int argument, but compiler have many options for that that's why it writes the error.

Reply:
I think the document https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#OpUConvert proves my gusses.

Reply:
> I think the document https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#OpUConvert proves my gusses.

I hardly see what it proves, from the spec `This is either a truncate or a zero extend.` whether the destination type is bigger or smaller, this is not a saturating operation: Uconvert 0x0F00 to uchar yields 0x0, SatConvertStoU yields 0xFF.

> that converts int to uint, it writes "call to '__spirv_UConvert_Ruint' is ambiguous"

This is expected as this operation does not exist in SPIR-V, so the overload does not exist. You are trying to do a 32 to 32 bits conversion (see https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#OpUConvert: `The component width cannot equal the component width in Result Type.`). The operation you are looking for in this cast is simply a `reinterpret_cast` in that case.

Reply:
Ok, I will remake